### PR TITLE
Fixed an issue with mounting a parcel when StrictMode is enabled

### LINKED ISSUE #160: Parcel never mounts component with StrictMode enabled
With React 18 development library and <React.StrictMode> the <Parcel> component does not mount the actual parcel. 

Observed flow:
* componentDidMount is called which calls addThingToDo
* componentWillUnmount is called which calls addThingToDo and immediately sets this.unmounted = true
* First addThingToDo is invoked and immediately returns as this.unmounted is true.

### PR DESCRIPTION
**Pull request by nickbosland**
[StrictMode](https://react.dev/reference/react/StrictMode) in React 18 renders a component an extra time where the component is unmounted and mounted again. This sets the `unmounted` value to `true` which prevent to component from actually mounting.

Fixes #160

### CODE REVIEW

**Comment by lipsumar** [comment]
We are anxiously waiting for this PR to land. We've tested it and it seems to work fine.

**Comment by MilanKovacic** [comment]
I will take a look.

**Comment by aaronbates** [comment]
Hi @MilanKovacic did you get a chance to take a look? We'd very much appreciate it!

File: src/parcel.test.js
Comment (MilanKovacic) [inline src/parcel.test.js:1]: Can we add a new test that renders in strict mode using the https://testing-library.com/docs/react-testing-library/api/#configure-option option, and leave existing tests as-is.
Reason for this is that strict mode alters the behaviour, and tests no longer test how library will behave in production.

**Review by MilanKovacic — CHANGES_REQUESTED** [review]
Reviewed

**Comment by aaronbates** [comment]
Thank you @MilanKovacic — appreciated.

@nickbosland if you get chance to take a look at the feedback that would be great, or equally we'd be happy to submit a PR if you're short on time. Thanks all

File: src/parcel.test.js
Comment (nickbosland) [reply src/parcel.test.js:1]: Makes sense, I have pushed the changes where I updated `@testing-library/react` to 13.4.0. This is the version that added the proposed configure function and I also updated the parcel test where it run all the test with and without strict mode enabled.

**Review by MilanKovacic — APPROVED** [review]
Looks good. Thank you.

**Comment by aaronbates** [comment]
@nickbosland are you OK to merge this if you have rights? TY

**Comment by lipsumar** [comment]
@MilanKovacic what are the next steps to get this released ?

**Comment by MilanKovacic** [comment]
It is waiting for approval from Joel.

File: src/parcel.js
Comment (joeldenning) [inline src/parcel.js:29]: I ran this code locally to see which tests failed in strict mode when this line was commented versus uncommented.

All tests pass regardless of whether this line of code is included. Please add a new test that demonstrates that `this.unmounted = false` is necessary.

**Review by joeldenning — CHANGES_REQUESTED** [review]
The code change makes sense. I've left one comment on the tests.

File: src/parcel.js
Comment (aaronbates) [reply src/parcel.js:29]: cc @nickbosland thanks

Comment (nickbosland) [reply src/parcel.js:29]: Thanks for the CC. I somehow missed the the notification. 

I have updated the `@testing-library/react` to [14.2.0](https://github.com/testing-library/react-testing-library/releases/tag/v14.2.0) which actually adds the option to configure whether strict mode is used in the tests or not. I also messed up the property name of the option somehow 🤦. I had to move the `configure` to the `beforeEach` as it does not work in the `beforeAll`.

Now multiple tests fail when this is line is removed.

Comment (aaronbates) [reply src/parcel.js:29]: Thanks @nickbosland — cc @joeldenning would you be happy to proceed with the PR now? Appreciate you taking a look!

Comment (lipsumar) [reply src/parcel.js:29]: @MilanKovacic @joeldenning any chance this can be reviewed again ?

Comment (MilanKovacic) [reply src/parcel.js:29]: I've contacted Joel to re-review it.

**Review by joeldenning — APPROVED** [review]

**Pull request merged**
